core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -614,6 +585,74 @@ private[worker] class Worker(
     webUi.stop()
     metricsSystem.stop()
   }
+
+  private def trimFinishedExecutorsIfNecessary(): Unit = {
+    if (finishedExecutors.size > retainedExecutors) {
+      finishedExecutors.take(math.max(finishedExecutors.size / 10, 1)).foreach {
```

Comment:
yes, the current design ensures that the actor is the channel getting WorkerStates

https://github.com/apache/spark/blob/069a4c414db4612d7bdb6f5615c1ba36998e5a49/core/src/main/scala/org/apache/spark/deploy/worker/ui/WorkerPage.scala#L36

https://github.com/apache/spark/blob/069a4c414db4612d7bdb6f5615c1ba36998e5a49/core/src/main/scala/org/apache/spark/deploy/worker/ui/WorkerPage.scala#L41

I'm adding the comments